src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java

```diff
@@ -0,0 +1,143 @@
+import edu.ucsb.cs156.gauchoride.errors.EntityNotFoundException;
+import edu.ucsb.cs156.gauchoride.repositories.DriverAvailabilityRepository;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+@Tag(name = "Driver Availability")
+@RequestMapping("/api/driverAvailability")
+@RestController
+
+public class DriverAvailabilityController extends ApiController{
+    
+    @Autowired
+    DriverAvailabilityRepository driverAvailabilityRepository;
+
+    @Operation(summary = "List all availabilities, only current user's if not admin")
+    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_DRIVER')")
+    @GetMapping("/all")
+    public Iterable<DriverAvailability> allAvailabilities() {
+        Iterable<DriverAvailability> availabilities;
+
+        if (getCurrentUser().getRoles().contains(new SimpleGrantedAuthority("ROLE_ADMIN"))) {
+            availabilities = driverAvailabilityRepository.findAll();
+        } else {
+            availabilities = driverAvailabilityRepository.findAllByDriverId(getCurrentUser().getUser().getId());
+        }
+
+        return availabilities;
+    }
+
+    @Operation(summary = "Get a single availability by id, can only be current user's if not admin")
+    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_DRIVER')")
+    @GetMapping("")
+    public DriverAvailability getById(
+            @Parameter(name="id", description = "long, Id of the availability to get", 
+            required = true)  
+            @RequestParam Long id) {
+                DriverAvailability availability;
+        
+        if (getCurrentUser().getRoles().contains(new SimpleGrantedAuthority("ROLE_ADMIN"))) {
+            availability = driverAvailabilityRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(DriverAvailability.class, id));;
+        } else {
+            availability = driverAvailabilityRepository.findByIdAndDriverId(id, getCurrentUser().getUser().getId())
+                .orElseThrow(() -> new EntityNotFoundException(DriverAvailability.class, id));
+        }
+
+        return availability;
+    }
+
+    @Operation(summary = "Create a new ride")
```

Comment:
Says `Create a new ride` but should be `Create a new driver availability` or similar